src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/adjourncase/AdjournCaseAboutToSubmitHandlerTest.java

```diff
@@ -63,53 +65,49 @@ public class AdjournCaseAboutToSubmitHandlerTest {
     private PreviewDocumentService previewDocumentService;
 
     private SscsCaseData sscsCaseData;
-    private AutoCloseable autoCloseable;
+
+    private void setUpAdjournCaseMocks() {
+        when(callback.getEvent()).thenReturn(EventType.ADJOURN_CASE);
+        when(callback.getCaseDetails()).thenReturn(caseDetails);
+        when(caseDetails.getCaseData()).thenReturn(sscsCaseData);
+    }
 
     @BeforeEach
-    public void setUp() {
-        autoCloseable = openMocks(this);
+    void setUp() {
         ReflectionTestUtils.setField(handler, "isAdjournmentEnabled", true);
 
-        when(callback.getEvent()).thenReturn(EventType.ADJOURN_CASE);
-        when(callback.getCaseDetails()).thenReturn(caseDetails);
         sscsCaseData = SscsCaseData.builder().ccdCaseId("ccdId")
             .appeal(Appeal.builder().build())
             .schedulingAndListingFields(SchedulingAndListingFields.builder()
                 .hearingRoute(HearingRoute.GAPS)
                 .build())
             .build();
-        when(caseDetails.getCaseData()).thenReturn(sscsCaseData);
-    }
-
-    @AfterEach
-    public void tearDown() throws Exception {
-        autoCloseable.close();
     }
 
     @DisplayName("Given a non adjourn case event, then return false")
     @Test
-    public void givenNonAdjournCaseEvent_thenReturnFalse() {
+    void givenNonAdjournCaseEvent_thenReturnFalse() {
         when(callback.getEvent()).thenReturn(EventType.APPEAL_RECEIVED);
         assertThat(handler.canHandle(ABOUT_TO_SUBMIT, callback)).isFalse();
     }
 
     @DisplayName("Given caseDetails is null, then return false")
     @Test
-    public void givenCaseDetailsIsNull_thenReturnFalse() {
-        when(callback.getCaseDetails()).thenReturn(null);
+    void givenCaseDetailsIsNull_thenReturnFalse() {
         assertThat(handler.canHandle(ABOUT_TO_SUBMIT, callback)).isFalse();
     }
 
     @DisplayName("Given caseData is null, then return false")
     @Test
-    public void givenCaseDataIsNull_thenReturnFalse() {
-        when(callback.getCaseDetails().getCaseData()).thenReturn(null);
+    void givenCaseDataIsNull_thenReturnFalse() {
+        lenient().when(callback.getCaseDetails()).thenReturn(caseDetails);
```

Comment:
Don't do lenient(), you should instead fix the tests to only have the correct mocks

Reply:
The problem here was that I need non-null caseDetails, and null caseData. The point is the if statement checks both conditions, and I want to test them independently. If I remove lenient, then Mockito cries about unnecessary stubbings. But how can I specifically test null caseData, without the if returning false at the first hurdle of null caseDetails?

Reply:
resolved by splitting into @Nested classes

Reply:
Replaced Nested classes with extracting separate test classes which extend a common abstract setup class